client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -533,17 +533,11 @@ public String getVersion() {
 
    @Override
    public <T> Query<T> query(String query) {
-      if (queryFactory == null) {
```

Comment:
This will just be a NPE now instead. I am not sure what this is supposed to be @tristantarrant can chime in.